api.go

```diff
@@ -186,9 +186,24 @@ func (s *Stats) Zero() bool {
 		s.Wait > 0)
 }
 
+// Progress contains information about the global progress of the running search query.
+// This is used by the frontend to reorder results and emit them when stable.
+type Progress struct {
+	// Priority of the shard that was searched.
+	ShardPriority float64
+
+	// Maximum priority of a shard that is being searched in parallel. This is used to reorder results
+	// when the result set is known to be stable-- that is, when a result's ShardPriority is greater than
+	// the max(MaxPendingShardPriority) from the latest results of each backend, it can be returned to the user.
+	//
+	// MaxPendingShardPriority decreases monotonically in each SearchResult.
+	MaxPendingShardPriority float64
+}
```

Comment:
I think we should remove "Shard" from the field names here, and from the description. The API we are communicating doesn't care about shards / etc, it cares about sorting of results. So really what we have here is a `Priority` for the current `SearchResult`, and the `MaxPendingPriority` for it. WDYT?